### PR TITLE
Adding Soli to Lohnsteuer test cases

### LINKED ISSUE #563: ENH: Tests for `soli_st_lohnst_m`
#150 implements Solidaritätszuschlag on withholding tax. The function is not necessarily correct and also untested.

### PR DESCRIPTION
**Pull request by JakobWegmann**
### What problem do you want to solve?

Adding Soli to the lohnsteuer test cases in GETTSIM, thereby closes [#563](https://github.com/iza-institute-of-labor-economics/gettsim/issues/563).

The actual work has been done in [gettsim-code-for-picking ](https://github.com/iza-institute-of-labor-economics/gettsim-code-for-picking/pull/3).

The PR also harmonizes the names of all lohnsteuer related things in GETSSIM to lohnst_m (instead of the sometimes used lohn_st_m).

### Todo

- [x] Document PR in CHANGES.md.

### CODE REVIEW

**Comment by JakobWegmann** [comment]
@hmgaudecker I'm a little confused about the naming convention right now. Should the files and variables be called lohnst or lohn_st?

I thought that the idea was to rename everything to lohnst. But soli is called soli_st and and income tax eink_st.

**Comment by hmgaudecker** [comment]
> @hmgaudecker I'm a little confused about the naming convention right now. Should the files and variables be called lohnst or lohn_st?
> 
> I thought that the idea was to rename everything to lohnst. But soli is called soli_st and and income tax eink_st.

That is because of this part of [GEP-01](https://gettsim.readthedocs.io/en/latest/geps/gep-01.html#abstract):

> Abbreviations of words that form a part of these names are always followed by an underscore, unless it is the last word.

So there is no abbreviation in the middle of `lohnst`, whereas there is one in `eink_st`. We should change `soli_st` eventually, I believe. Thought back then that the `_st` would be good for signalling that it is in the realm of taxes, but at least I have changed mz mind. That'll be for another day, however.

**Comment by JakobWegmann** [comment]
> That is because of this part of [GEP-01](https://gettsim.readthedocs.io/en/latest/geps/gep-01.html#abstract):
> 
> > Abbreviations of words that form a part of these names are always followed by an underscore, unless it is the last word.
> 
> So there is no abbreviation in the middle of `lohnst`, whereas there is one in `eink_st`. 

Ahh, that makes sense!

**Review by MImmesberger — APPROVED** [review]
Looks good, no complaints from my side!

File: CHANGES.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ releases are available on [Anaconda.org](https://anaconda.org/conda-forge/gettsi
   ({ghuser}`ChristianZimpelmann`).
 - {gh}`583` Automatic conversion between different time units ({ghuser}`lars-reimann`).
 - {gh}`581` Add `_y` suffix to names of yearly variables ({ghuser}`lars-reimann`).
+- {gh}`653` Adding Soli to Lohnsteuer test cases ({ghuser}`JakobWegmann`).
```
Comment (hmgaudecker) [inline CHANGES.md:41]: Add on top, please.

**Review by hmgaudecker — APPROVED** [review]
Looks great, thanks!

**Pull request merged**